PIPReader/src/main/java/it/cnr/iit/ucs/pipreader/PIPReader.java

```diff
@@ -55,7 +58,7 @@
 public final class PIPReader extends PIPBase {
 
     private static Logger log = Logger.getLogger( PIPReader.class.getName() );
-    private PIPJournalHelper journal;
+    private Optional<JournalingInterface> journal;

```

Comment:
we should use optional only in initialization or hide the fact that the journal is disabled inside the journal